storey/flow.py

```diff
@@ -1593,36 +1780,13 @@ def select_runnables(self, event) -> Optional[Union[list[str], list[ParallelExec
 
     def _init(self):
         super()._init()
-        num_processes = 0
-        num_threads = 0
-        mp_context = multiprocessing.get_context("spawn")
         for runnable in self.runnables:
-            if runnable.name in self._runnable_by_name:
-                raise ValueError(f"ParallelExecutionRunnable name '{runnable.name}' is not unique")
-            runnable.init()
-            self._runnable_by_name[runnable.name] = runnable
-            if runnable.execution_mechanism == "process_pool":
-                num_processes += 1
-            elif runnable.execution_mechanism == "dedicated_process":
-                self._process_executor_by_runnable_name[runnable.name] = ProcessPoolExecutor(
-                    max_workers=1,
-                    mp_context=mp_context,
-                    initializer=_set_global,
-                    initargs=(runnable,),
-                )
-            elif runnable.execution_mechanism == "thread_pool":
-                num_threads += 1
-            elif runnable.execution_mechanism not in ("asyncio", "naive"):
-                raise ValueError(f"Unsupported execution mechanism: {runnable.execution_mechanism}")
-
-        # enforce max
-        num_threads = min(num_threads, self.max_threads)
-
-        self._executors = {}
-        if num_processes:
-            self._executors["process_pool"] = ProcessPoolExecutor(max_workers=num_processes, mp_context=mp_context)
-        if num_threads:
-            self._executors["thread_pool"] = ThreadPoolExecutor(max_workers=num_threads)
+            if runnable.execution_mechanism == ParallelExecutionMechanisms.shared_executor:
+                self.context.executor.init_runnable(runnable=runnable.name)
+            else:
+                self.runnable_executor.add_runnable(runnable=runnable)
```

Comment:
Won't this fail with `ParallelExecutionRunnable name '{runnable.name}' is not unique` if the same runnable is used from different runners (which is what we're trying to support)? Or am I missing something?